lib/grammers-client/src/client/files.rs

```diff
@@ -223,71 +234,78 @@ impl Client {
         };
 
         let big_file = size > BIG_FILE_SIZE;
-        let mut buffer = vec![0; MAX_CHUNK_SIZE as usize];
-        let total_parts = ((size + buffer.len() - 1) / buffer.len()) as i32;
-        let mut md5 = md5::Context::new();
-
-        for part in 0..total_parts {
-            let mut read = 0;
-            while read != buffer.len() {
-                let n = stream.read(&mut buffer[read..]).await?;
-                if n == 0 {
-                    if part == total_parts - 1 {
-                        break;
-                    } else {
-                        return Err(io::Error::new(
-                            io::ErrorKind::UnexpectedEof,
-                            "reached EOF before reaching the last file part",
-                        ));
+        let parts = Arc::new(PartStream::new(stream, size));
+        let total_parts = parts.total_parts();
+
+        if big_file {
+            let mut tasks = Vec::with_capacity(WORKER_COUNT);
+            for _ in 0..WORKER_COUNT {
+                let handle = self.clone();
+                let parts = Arc::clone(&parts);
+                let task = tokio::task::spawn(async move {
```

Comment:
This `spawn` is somewhat unfortunate, as nowhere else in the code tasks are spawned. Would it be possible to do without? I'm trying to avoid it because it's executor-specific, and I'd like to move to a `futures`-only approach at some point (or the best I could).

Reply:
Hmm, how do you want to do that without spawning tasks?

Reply:
Done.